### PR TITLE
Wrap error with more context when pinning contents

### PR DESCRIPTION
**Pull request by en0ma**
When looking at the logs on Grafana, I notice we have so many `context canceled` errors when pinning contents.

Screenshot 2022-05-10 at 22 11 29

Currently, It is hard to tell why this is happening. This PR makes some changes to wrap the errors with additional context, so we can easily trace what is going on, so as to provide a fix if one is required.

### CODE REVIEW

File: pinning.go [hunk 1]
```diff
@@ -121,16 +121,19 @@ func (s *Server) doPinning(ctx context.Context, op *pinner.PinningOperation, cb
...
-	// this provide call goes out immediately
-	if err := s.Node.FullRT.Provide(ctx, op.Obj, true); err != nil {
-		log.Infof("provider broadcast failed: %s", err)
-	}
-
-	// this one adds to a queue
-	if err := s.Node.Provider.Provide(op.Obj); err != nil {
-		log.Infof("providing failed: %s", err)
-	}
+	go func() {
```
Comment (en0ma) [inline pinning.go:124]: no need to block for these

**Review by alvin-reyes — COMMENTED** [review]
Great stuff. I noticed that we definitely need more informative logging in estuary and this PR definitely helps with tracing errors.

File: pinning.go (hunk 1 above)
Comment (en0ma) [reply pinning.go:124]: noted

File: cmd/estuary-shuttle/main.go [hunk 2]
```diff
@@ -1124,18 +1125,19 @@ func (s *Shuttle) Provide(ctx context.Context, c cid.Cid) error {
...
 	if s.Node.FullRT.Ready() {
 		if err := s.Node.FullRT.Provide(subCtx, c, true); err != nil {
-			return fmt.Errorf("failed to provide newly added content: %w", err)
+			return errors.Wrap(err, "failed to provide newly added content")
 		}
 	} else {
 		log.Warnf("fullrt not in ready state, falling back to standard dht provide")
 		if err := s.Node.Dht.Provide(subCtx, c, true); err != nil {
-			return fmt.Errorf("fallback provide failed: %w", err)
+			return errors.Wrap(err, "fallback provide failed")
```
[the remark this answers is not in the document]
Comment (en0ma) [reply cmd/estuary-shuttle/main.go:1133]: @whyrusleeping the current error does not have the stacktrace. I can't debug where exactly we are getting those error from

File: pinning.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (en0ma) [reply pinning.go]: reverted

File: cmd/estuary-shuttle/main.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (en0ma) [reply cmd/estuary-shuttle/main.go:1133]: @whyrusleeping I see what you mean. The logger, in its current settings, will not log the StackTrace. I just played around with it a bit and got some trace. I just need to clean it up a bit and I will push the changes

[the remark this answers is not in the document]
Comment (en0ma) [reply cmd/estuary-shuttle/main.go:1133]: Ok, I finally got this to work - it needed the verbose (%+v) flag to log the stackTrace if one exists (`errors.Wrap` and `xerror.Errorf` produces one). 

@whyrusleeping, they are not the same. `errors.Wrap` and `xerror.Errorf` are almost the same, but `errors.Wrap` is more ordered. See sample output for all 3 approaches;

- fmt.Errorf - no stackTrace
Screenshot 2022-05-12 at 11 47 23

- xerror.Errorf - has stackTrace, but not ordered
Screenshot 2022-05-12 at 12 28 12

- errors.Wrap - has stackTrace and ordered
Screenshot 2022-05-12 at 12 41 05
cc: @alvin-reyes

File: cmd/estuary-shuttle/main.go [hunk 3]
```diff
@@ -1458,9 +1460,8 @@ func (d *Shuttle) doPinning(ctx context.Context, op *pinner.PinningOperation, cb
 	*/
 
 	if err := d.Provide(ctx, op.Obj); err != nil {
-		log.Warn(err)
+		return errors.Wrapf(err, "failed to Provide - contID(%d), cid(%s)", op.ContId, op.Obj.String())
```
Comment (gmelodie) [inline cmd/estuary-shuttle/main.go:1463]:
```suggestion
		return errors.Wrapf(err, "failed to provide - contID(%d), cid(%s)", op.ContId, op.Obj.String())
```

File: cmd/estuary-shuttle/main.go [hunk 4]
```diff
@@ -1551,7 +1552,7 @@ func (d *Shuttle) addDatabaseTrackingToContent(ctx context.Context, contid uint,
 		return node.Links(), nil
 	}, root, cset.Visit, merkledag.Concurrent())
 	if err != nil {
-		return err
+		return errors.Wrap(err, "failed to Walk DAG")
```
Comment (gmelodie) [inline cmd/estuary-shuttle/main.go:1555]:
```suggestion
		return errors.Wrap(err, "failed to walk DAG")
```

**Review by gmelodie — APPROVED** [review]
Looking good! We need this!!

**Pull request merged**